recipes/libxcrypt/meta.yaml

```diff
@@ -0,0 +1,73 @@
+{% set name = "libxcrypt" %}
+{% set version = "4.4.28" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/besser82/libxcrypt/releases/download/v{{ version }}/libxcrypt-{{ version }}.tar.xz
+  sha256: 9e936811f9fad11dbca33ca19bd97c55c52eb3ca15901f27ade046cc79e69e87
+
+build:
+  number: 0
+  skip: true  # [win]
+  run_exports:
+    # The ABI is very stable
+    - {{ pin_subpackage('libxcrypt', max_pin=None) }}
+  script_env:
+    - OBSOLETE_API=no
+  script: $RECIPE_DIR/do_build.sh
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - make
+    - pkg-config
+    - perl
+
+test:
+  commands:
+    - test -f "${PREFIX}/include/crypt.h"
+    - test -f "${PREFIX}/lib/libcrypt${SHLIB_EXT}"
+    - test -f "${PREFIX}/lib/libcrypt.so.2"  # [not osx]
+    - test -f "${PREFIX}/lib/libcrypt.2.dylib"  # [osx]
+    - test -f "${PREFIX}/lib/pkgconfig/libxcrypt.pc"
+    - test ! -f "${PREFIX}/lib/libcrypt.so.1"
+    - test ! -f "${PREFIX}/lib/libcrypt.1.dylib"
+
+outputs:
+  - name: libxcrypt-1
+    script: do_build.sh
+    build:
+      skip: true  # [not linux]
+      script_env:
+        - OBSOLETE_API=glibc
+
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - make
+        - pkg-config
+        - perl
+
+    test:
+      commands:
+        - test ! -f "${PREFIX}/include/crypt.h"
+        - test ! -f "${PREFIX}/lib/libcrypt${SHLIB_EXT}"
+        - test ! -f "${PREFIX}/lib/pkgconfig/libxcrypt.pc"
+        - test -f "${PREFIX}/lib/libcrypt.so.1"
```

Comment:
With libnsl, we only packaged the new ABI `libnsl.so.2` and rebuilt packages. Don't we need to do the same here?

Reply:
I was thinking of making `libcrypt.so.1` available in a standalone package so we can patch the repodata for old versions.

Reply:
To be clear, I mean in addition to rebuilding against `libcrypt.so.2`.

Reply:
Adding in new packages as a repodata patch was problematic last I checked as conda will generate warnings saying the environment is inconsistent.

Reply:
Ah that's annoying, I'll have a play and follow up in https://github.com/conda-forge/perl-feedstock/issues/56. Worst case installing `libxcrypt-1` can become the documented workaround to this issue.

Do you have any thoughts on the rest of this recipe?

Reply:
Makes sense